hibernate-core/src/main/java/org/hibernate/id/enhanced/NoopOptimizer.java

```diff
@@ -38,12 +38,6 @@ public Serializable generate(AccessCallback callback) {
 		// The lastSourceValue field is only accessed by tests,
 		// so this is not a concern.
 		IntegralDataTypeHolder value = callback.getNextValue();
```

Comment:
After discussing further with Stephen Fikes, maybe it would be better to not make this change, since it's been done this way for such a long time. We could just log a warning indicating that Hibernate will cycle the sequence until it gets to 1. 

Do you have a preference one way or the other?

I would still make the other changes so that a negative allocation size can be used with negative sequence values.